packages/desktop-client/src/components/FloatableSidebar.js

```diff
@@ -36,30 +35,31 @@ export function SidebarProvider({ children }) {
 }
 
 export function useSidebar() {
-  return useContext(SidebarContext);
+  useViewportSize(); // Force re-render on window resize
+  let windowWidth = document.documentElement.clientWidth;
+  let alwaysFloats = windowWidth < 668;
+
+  let context = useContext(SidebarContext);
+  return { ...context, alwaysFloats };
 }
 
 function Sidebar({ floatingSidebar }) {
   let [hidden, setHidden] = useState(true);
   let sidebar = useSidebar();
 
-  let windowWidth = useViewportSize().width;
-  let sidebarShouldFloat = floatingSidebar || windowWidth < breakpoints.medium;
-
-  if (!sidebarShouldFloat && hidden) {
-    setHidden(false);
-  }
+  let sidebarShouldFloat = floatingSidebar || sidebar.alwaysFloats;
 
   useEffect(() => {
     let cleanups = [
       sidebar.on('show', () => setHidden(false)),
       sidebar.on('hide', () => setHidden(true)),
-      sidebar.on('toggle', () => setHidden(hidden => !hidden)),
+      // using hidden => !hidden causes a bug bc the handler is called twice?!?!
+      sidebar.on('toggle', () => setHidden(!hidden)),
```

Comment:
I don't quite understand the need for this change. Cloned locally and reverted it and everything seemed to function just fine. What issue were you seeing?

Reply:
I think the repro was (at least on iPhone):

1. Tap the menu button to show the sidebar
2. Tap on the sidebar somewhere
3. Tap the menu button again

It wouldn’t hide the sidebar.

Reply:
I think it's the `onMouseOver` on [line 82](https://github.com/actualbudget/actual/pull/868/files#diff-a63f6532771a7d1f709ef9baa8e94cb8b3e40ceebd821a4f7a1e5d6c8eddb65fL82) interfering here and causing issues.

Can we get rid of all the custom logic and make the sidebar simple (& stupid)? Maintaining all this very complicated logic with various edge-cases is otherwise going to be a nightmare.

Reply:
Bump on this. I still think the show/hide logic is still too brittle. Maintaining it will be a nightmare. Instead of trying to hack in some more functionality in it - could we instead refactor it to simplify it?

Reply:
Updated! I think it’s as simple as it can get now.

Reply:
I think it is important to keep the hover behavior so accessing the sidebar remains a lightweight interaction without a bunch of clicking.